openapi2kong/validator.go

```diff
@@ -72,17 +73,23 @@ func generateParameterSchema(operation *v3.Operation, insoCompat bool) []map[str
 				paramConf["required"] = false
 			}
 
-			schema := extractSchema(parameter.Schema)
+			schema, schemaMap := extractSchema(parameter.Schema)
 			if schema != "" {
 				paramConf["schema"] = schema
+
+				_, typeStr, ok := fetchOneOfAndType(schemaMap)
+				if ok && typeStr == "" {
+					return nil,
+						fmt.Errorf(`parameter schemas for request-validator plugin using oneOf must have a top-level type property`)
```

Comment:
This is not only for `oneOf`, `anyOf` has the same problem for instance.

```suggestion
						fmt.Errorf(`parameter schemas for request-validator plugin must have a top-level type property`)
```

or

```suggestion
						fmt.Errorf(`parameter schemas for request-validator plugin must have a top-level type property (to know how to deserialize the parameter)`)
```